### PR TITLE
Add a fun type

### PR DESCRIPTION
**Pull request by yutopp**

### CODE REVIEW

File: test/test_spec.ml [hunk 1]
```diff
@@ -17,8 +17,7 @@ let%expect_test "test_spec.beam" =
             1
             ((
               TyContFun 6
-              (TyPredef 6 fun (
-                (TyProduct 6 ((TyVar 6 A))) (TyPredef 6 integer ())))
+              (TyFun 6 (TyProduct 6 ((TyVar 6 A))) (TyPredef 6 integer ()))
```
Comment (yutopp) [inline test/test_spec.ml]: NOTE: Now, `fun type` is no longer predefined type.

File: test/test_spec.ml [hunk 2]
```diff
@@ -35,7 +34,7 @@ let%expect_test "test_spec.beam" =
             1
             ((
               TyContFun 10
-              (TyPredef 10 fun ((TyProduct 10 ((TyVar 10 A))) (TyVar 10 B)))
+              (TyFun 10 (TyProduct 10 ((TyVar 10 A))) (TyVar 10 B))
```
Comment (yutopp) [inline test/test_spec.ml]: Same as above.

File: lib/abstract_format.ml [hunk 3]
```diff
@@ -113,13 +113,16 @@ and type_t =
   | TyAnyMap of line_t
   | TyMap of line_t * type_assoc_t list
   | TyVar of line_t * string
+  | TyFunAny of line_t
+  | TyFunAnyArity of line_t * type_t * type_t
```
Comment (amutake) [inline lib/abstract_format.ml]: I think it is enough to be `TyFunAnyArity of line_t * line_t * type_t` like [ExprRemoteCall](https://github.com/yutopp/obeam/blob/c1f65123e2474ff5cc2b221ebf72a8468c7a53b6/lib/abstract_format.ml#L66). What do you think about it?

Comment (yutopp) [reply lib/abstract_format.ml]: I selected `type_t` to clarify meaning of it, however `line_t` is also good if there have been already some examples which are using this interface. I'll fix it.

Comment (yutopp) [reply lib/abstract_format.ml]: Fixed b1c5479

**Comment by yutopp** [comment]
:raising_hand_woman:

**Pull request merged**